src/python/pants/engine/internals/selectors.py

```diff
@@ -56,12 +56,79 @@ class AwaitableConstraints:
     input_type: type
     is_effect: bool
 
+    # Parse the following Get forms:
+    #  c = await Get[C](B, b) # we can do this now
```

Comment:
Ah, so: this aspect of the description was stale over there. We actually only parse `Get(C, B, b)`, etc: while the indexing for type hints is supported, it's usually inferred, and the actual input type is required currently.

We briefly did actually support index-only (`Get[C](B, b)`), but we removed it. Sorry for the confusion.

Reply:
Is there still an intent to have the other forms that use brackets though?

That is:

```
await Get[E](Params((C, c), (D, d))) 
await Get[F](Params(C(...), D(...)))
```